src/ResultList.php

```diff
@@ -167,29 +180,56 @@ public function toArray()
             $needed = array();
             $retrieved = array();
 
-            if (is_array($found) || $found instanceof \ArrayAccess) {
+            if (is_array($found) || $found instanceof ArrayAccess) {
                 foreach ($found as $item) {
-                    $type = $item->getType();
+                    $type = isset($item->{Searchable::TYPE_FIELD}[0])
+                      ? $item->{Searchable::TYPE_FIELD}[0]
+                      : false;
 
+                    if (empty($type)) {
+                        throw new LogicException("type field not available");
+                    }
                     if (!array_key_exists($type, $needed)) {
-                        $needed[$type] = array($item->getId());
-                        $retrieved[$type] = array();
+                        $needed[$type] = [$item->getId()];
+                        $retrieved[$type] = [];
                     } else {
                         $needed[$type][] = $item->getId();
                     }
                 }
 
                 foreach ($needed as $class => $ids) {
-                    foreach ($class::get()->byIDs($ids) as $record) {
+                    foreach (DataObject::get($class)->byIDs($ids) as $record) {
                         $retrieved[$class][$record->ID] = $record;
                     }
                 }
 
                 foreach ($found as $item) {
                     // Safeguards against indexed items which might no longer be in the DB
-                    if (array_key_exists($item->getId(), $retrieved[$item->getType()])) {
-                        $this->resultsArray[] = $retrieved[$item->getType()][$item->getId()];
+                    $type = isset($item->{Searchable::TYPE_FIELD}[0])
+                      ? $item->{Searchable::TYPE_FIELD}[0]
+                      : false;
+
+                    if (empty($type)) {
+                        throw new LogicException("type field not available");
+                    }
+                    $id = $item->getId();
+                    if (!isset($retrieved[$type][$id])) {
+                        continue;
```

Comment:
A few newlines around here needed, must to break things up.

Reply:
Fixed